takari-lifecycle-plugin/src/main/java/io/takari/maven/plugins/compile/AbstractCompiler.java

```diff
@@ -151,6 +155,22 @@ public void setTransitiveDependencyReference(AccessRulesViolation transitiveDepe
     this.transitiveDependencyReference = transitiveDependencyReference;
   }
 
+  protected AccessRulesViolation getUnusedDeclaredDependency() {
+    return unusedDeclaredDependency;
+  }
+
+  public void setUnusedDeclaredDependency(AccessRulesViolation unusedDeclaredDependency) {
+    this.unusedDeclaredDependency = unusedDeclaredDependency;
+  }
+
+  protected Set<File> getReferencedClasspathEntries() {
```

Comment:
I'd define `getReferencedClasspathEntries()` as abstract here and throw `UnsupportedOperationException` in Javac compiler backend.